python/pip_install/pip_repository.bzl

```diff
@@ -367,6 +392,40 @@ whl_library = repository_rule(
     doc = """
 Download and extracts a single wheel based into a bazel repo based on the requirement string passed in.
 Instantiated from pip_repository and inherits config options from there.""",
-    implementation = _impl_whl_library,
+    implementation = _whl_library_impl,
     environ = common_env,
 )
+
+def package_annotation(
+        build_content = None,
+        copy_files = {},
+        copy_executables = {},
+        data = [],
+        data_exclude_glob = [],
+        srcs_exclude_glob = []):
+    """Annotations to apply to the BUILD file content from package generated from a `pip_repository` rule.
+
+    [cf]: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md
+
+    Args:
+        build_content (str, optional): Raw text to add to the generated `BUILD` file of a package.
+        copy_files (dict, optional): A mapping of `src` and `out` files for [@bazel_skylib//rules:copy_file.bzl][cf]
+        copy_executables (dict, optional): A mapping of `src` and `out` files for
+            [@bazel_skylib//rules:copy_file.bzl][cf]. Targets generated here will also be flagged as
+            executable.
+        data (list, optional): A list of labels to add as `data` dependencies to the generated `py_library` target.
+        data_exclude_glob (list, optional): A list of exclude glob patterns to add as `data` to the generated
+            `py_library` target.
+        srcs_exclude_glob (list, optional): A list of labels to add as `srcs` to the generated `py_library` target.
+
+    Returns:
+        str: A json encoded string of the provided content.
+    """
+    return json.encode(struct(
+        build_content = build_content,
```

Comment:
Could we also supply a label of a BUILD file in the repo here? Similar to `http_archive` (https://docs.bazel.build/versions/main/repo/http.html#http_archive-build_file) etc. It might be a nice way to completely override the BUILD file for a requirement.

Reply:
Ah, this is actually a misleading parameter. `build_content` is really `additive_build_content`. It's a way for us to add whatever we want to the generated BUILD file. Do you think there's a big use case for this? I have yet to come across it so I would opt out of adding it to this PR.

Reply:
If you haven't hit it then don't add it, we can always add it on later. Some folks may like the content in a file rather than inline however, but it's not a blocker imo.